##  Descriptive Statistics
Descriptive Statistics is the part of statistics that is 
utilized for summarizing,organizing,and describing important 
characteistics of a dataset.The proccess of getting insight into
the information through descriptive statistics involves the use
of central tendency measures (mean,median,mode) and measures of
variabilliy(range,variance,standard deviation,IQR).

## Measures of Central Tendency
## (Mean),(Median),(Mode)

## Mean
we can find the mean by adding all the values in the dataset,then divide vy the total numbers of values.

## Median
after arrange the data in ascending order if the number is even
the median is the average of the two middle values and if the number is odd the median is the middle value.

## Mode
It's the value that appears most frequently in the dataset.It can be one or more or none.

In [1]:
import numpy as np

In [3]:
data=np.array([20,12,15,12,40])
print(np.mean(data))
print(np.median(data))

19.8
15.0


## Measures of Spread
## (Range),(Variance),(Standard deviation),(Interquartile range(IQR)).

## Range
maximum value-minimum value

## Variance
indicates how far data values are from the mean.It's computed by taking the average of the sequares of the differences of the values from the mean.

## Standard deviation (SD)
indicates the spread of data around the mean. It is equal to the sequare root of the variance.If the (SD) is small the data is clustered around the mean.On the other hand, a large (SD) implies that data is widely spread.

## interquartile range (IQR)
indicates the spread of the middle 50% of the data. The steps for computation are as follow:
1- arrange the data increasing order
2- compute Q1 (first quartile)
3- compute Q3 (third quartile)
4- IQR=(Q3-Q1) it is not sensitive to outliers like the range.

In [5]:
data2=np.array([10,20,40,12,100])
print(np.std(data2))
print(np.var(data2))
q1,q3=np.percentile(data2,[25,75])
print(q3-q1)

33.52372294361114
1123.84
28.0


## Hands-On Lab: Describing a Real Dataset

In [6]:
import pandas as pd

In [9]:
df=pd.read_csv("tested.csv")
df.info()
df.head()

<class 'pandas.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    str    
 4   Sex          418 non-null    str    
 5   Age          332 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    str    
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     str    
 11  Embarked     418 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 39.3 KB


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,0,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,1,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,0,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q
3,895,0,3,"Wirz, Mr. Albert",male,27.0,0,0,315154,8.6625,NaN,S
4,896,1,3,"Hirvonen, Mrs. Alexander (Helga E Lindqvist)",female,22.0,1,1,3101298,12.2875,NaN,S


In [20]:
data3=df["Fare"].dropna()
mean=np.mean(data3)
print("mean:",mean)
median=np.median(data3)
print("median:",median)
mode=data3.mode()[0]
print("Mode:",mode)
sd=np.std(data3)
print("standard deviation:",sd)
Q1=np.percentile(data3,25)
Q3=np.percentile(data3,75)
iqr=Q3-Q1
print("IQR:",iqr)


mean: 35.627188489208635
median: 14.4542
Mode: 7.75
standard deviation: 55.840500479541035
IQR: 23.6042


## Wich measure better represents a  typical value?
The median provides a better representation of the typical value
for the Fare since there are outliers that make the average higher. The median is not influenced by outliers and therefor provides an accurate representation of the typical value.

## Summary
The Fare column has a mean value of 35.627 and a median value of 14.4542.The mean and the median show that the data is skewed because of the presence of some high fare values.The mode value is 7.75 and it depicts the most common value in the dataset.
The median is a better indicator of the central tendency because it is less affected by outliers.